pkg/nats/nats.go

```diff
@@ -0,0 +1,187 @@
+package nats
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	natslib "github.com/nats-io/nats.go"
+	sourcesdk "github.com/numaproj/numaflow-go/pkg/sourcer"
+	"go.uber.org/zap"
+
+	"numaflow-nats-source/pkg/config"
+	"numaflow-nats-source/pkg/utils"
+)
+
+const (
+	defaultBufferSize = 1000
+)
+
+type Message struct {
+	payload    string
+	readOffset string
+	id         string
+}
+
+type natsSource struct {
+	natsConn *natslib.Conn
+	sub      *natslib.Subscription
+
+	bufferSize int
+	messages   chan *Message
+
+	volumeReader utils.VolumeReader
+
+	logger *zap.Logger
+}
+
+type Option func(*natsSource) error
+
+// WithLogger is used to return logger information
+func WithLogger(l *zap.Logger) Option {
+	return func(o *natsSource) error {
+		o.logger = l
+		return nil
+	}
+}
```

Comment:
I don't think this is being used

Reply:
After a second thought, I am going to make a change to use it New.